### PR TITLE
Protobuf serializer for Exceptions

### PR DESCRIPTION
**Pull request by alexvaluyskiy**

### CODE REVIEW

File: src/core/Akka/Actor/Exceptions.cs [hunk 1]
```diff
@@ -184,13 +182,21 @@ public ActorInitializationException(IActorRef actor, string message, Exception c
         protected ActorInitializationException(SerializationInfo info, StreamingContext context)
             : base(info, context)
         {
+            Actor = (IActorRef)info.GetValue("Actor", typeof(IActorRef));
+        }
+
+        public override void GetObjectData(SerializationInfo info, StreamingContext context)
```
Comment (alexvaluyskiy) [inline src/core/Akka/Actor/Exceptions.cs]: It is mandatory method for exceptions, which contains custom properties

**Comment by Aaronontheweb** [comment]
@alexvaluyskiy I'm getting compilation failures here since I rebased

File: src/core/Akka.Remote.Tests/RemoteRouterSpec.cs [hunk 2]
```diff
@@ -302,7 +302,7 @@ public void RemoteRouter_must_let_remote_deployment_be_overridden_by_remote_conf
             masterSystem.Stop(router);
         }
 
-        [Fact(Skip = "Serialization of custom deciders is currently not supported")]
+        [Fact]
         public void RemoteRouter_must_set_supplied_SupervisorStrategy()
```
Comment (alexvaluyskiy) [inline src/core/Akka.Remote.Tests/RemoteRouterSpec.cs]: This test is working now

File: src/core/Akka/Actor/Exceptions.cs [hunk 3]
```diff
@@ -134,12 +134,10 @@ public class ActorInterruptedException : AkkaException
     /// </summary>
     public class ActorInitializationException : AkkaException
     {
-        private readonly IActorRef _actor;
-
         /// <summary>
         /// Initializes a new instance of the <see cref="ActorInitializationException"/> class.
         /// </summary>
-        protected ActorInitializationException()
+        public ActorInitializationException()
```
Comment (alexvaluyskiy) [inline src/core/Akka/Actor/Exceptions.cs]: public constructor is mandatory for NetCore support (due to limitations of our serializer)

**Comment by alexvaluyskiy** [comment]
@heynickc MNTK tests don't want to pass. I don't know why

**Comment by Aaronontheweb** [comment]
@alexvaluyskiy looks like @Arkatufus has fixed it here https://github.com/akkadotnet/akka.net/pull/2936

**Pull request merged**